packages/backend/src/utils/SlugUtils.ts

```diff
@@ -78,11 +82,6 @@ export const generateUniqueSlugScopedToProject = async (
                 'generateUniqueSlugScopedToProject',
             );
     }
-    const baseSlug = generateSlug(name);
-    const matchingSlugs: string[] = await matchingSlugsQuery
-        .select(`${SavedChartsTableName}.slug`)
```

Comment:
Moved this inside the switch case since the select and filters change based on initial query. eg: for dashboards we need to select and filter the dashboard table.